include/BALL/KERNEL/bond.iC

```diff
@@ -32,12 +32,20 @@ void Bond::destroy()
 BALL_INLINE
 void Bond::finalize()
 {
-	if (*first_ > *second_)
-	{
-		Atom* tmp = first_;
-		first_ = second_;
-		second_ = tmp;
-	}
+        if (first_ && second_)
+        {
+                if (*first_ > *second_)
+                {
+                        Atom* tmp = first_;
+                        first_ = second_;
+                        second_ = tmp;
+                }
+        }
+        else
```

Comment:
I came upon this problem because XDRPersistenceManager_test (test [EXTRA] full_test0, line 424) goes into a Segfault in Xcode7 setups because of this function.

I agree with you that bonds without atoms do not make sense. However, the default ctor of the Bond class sets both of its atoms to zero, which is used in this test. Following the argumentation that such a bond doesn't make sense, this test is simply wrong and should either be deleted or fixed, right?